rust-symcrypt/src/hmac.rs

```diff
@@ -481,16 +515,38 @@ pub struct HmacSha256State {
     // Using an `HmacSha256Inner` state that is Pin<Box<>>'d. Memory allocation is not handled by SymCrypt and Self is moved
     // around when returning from `HmacSha256State::new()`. Box<> heap allocates the memory and ensures that it does not move
     // within its lifetime.
-    inner: Pin<Box<HmacSha256Inner>>,
+    state: Pin<Box<HmacSha256Inner>>,
+
+    // Must Arc<> the expanded_key field since it must be ref counted, clones of HmacSha265State will reference the same expanded key.
+    // Arc<T> pointer can move, but its reference T is Pin<Box<>>'d to ensure that the address does not move during its lifetime.
+    key: Arc<Pin<Box<HmacSha256ExpandedKey>>>,
```

Comment:
I think a `Pin<Arc<HmacSha256ExpandedKey>>` would likely work here.

One snag you might run into is that you can't generally mutate the inside of an `Arc` without a `Mutex`. But, if you have the pattern of needing to initialize it before you share it and then it becomes immutable, you can use [`Arc::get_mut()`](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.get_mut).

Reply:
I don't see any reason that they would ever need to modify the inside of the `Arc` in this case. The arc is used to ref count the usage of the key so I think it should be fine like this. I will keep the `Arc::get_mut()` in mind for the future